base/base/Decimal_fwd.h

```diff
@@ -15,6 +15,20 @@ using UInt128 = wide::integer<128, unsigned>;
 using Int256 = wide::integer<256, signed>;
 using UInt256 = wide::integer<256, unsigned>;
 
+template <typename T>
```

Comment:
L. 13-16 (this file) and l. 8-11 (extended_types.h) define the typedefs. I guess that's unrelated but maybe we can define the types just once?